vm/native/native.cpp

```diff
@@ -69,7 +70,9 @@ extern "C" {
   }
 
   uintptr_t gcMallocAtomic(int32_t size) {
-    return (uintptr_t)GC_MALLOC_ATOMIC(size);
+    uintptr_t ptr = (uintptr_t)GC_MALLOC_ATOMIC(size);
+    memset((void*)ptr, 0, size);
```

Comment:
> Nit: we could subtract the size of the header here because we don't need to zero it out, but I guess it doesn't make a big difference.

Yeah, I suspect it doesn't make much difference, and it would complicate the implementation, since then gcMallocAtomic would need to know the header size. Just for kicks, I tried it over in this other branch, and it didn't seem to make a difference:

https://github.com/mykmelez/pluotsorbet/compare/intex-fill-new-primitive-array...mykmelez:intex-fill-new-primitive-array-offset